stubs/uWSGI/uwsgi.pyi

```diff
@@ -0,0 +1,241 @@
+def cache_mul(__key: str | bytes, __factor: int = 2, __expires: int = 0, __cache_name: str = ...) -> _TrueOrNone: ...
+def cache_num(__key: str | bytes, __cache_name: str = ...) -> int: ...
+def cache_set(
+    __key: str | bytes, __value: str | bytes | ReadOnlyBuffer, __expires: int = 0, __cache_name: str = ...
+) -> _TrueOrNone: ...
+def cache_update(
+    __key: str | bytes, __value: str | bytes | ReadOnlyBuffer, __expires: int = 0, __cache_name: str = ...
+) -> _TrueOrNone: ...
+def queue_get(__index: int) -> bytes | None: ...
+def queue_set(__index: int, __message: str | bytes | ReadOnlyBuffer) -> _TrueOrNone: ...
+@overload
+def queue_last(__num: Literal[0] = 0) -> bytes | None: ...  # type:ignore[misc]
+@overload
+def queue_last(__num: int) -> list[bytes | None]: ...
+def queue_push(__message: str | bytes | ReadOnlyBuffer) -> _TrueOrNone: ...
+def queue_pull() -> bytes | None: ...
+def queue_pop() -> bytes | None: ...
+def queue_slot() -> int: ...
+def queue_pull_slot() -> int: ...
+def snmp_set_community(__snmp_community: str) -> Literal[True]: ...
+def snmp_set_counter32(__oid_num: int, __value: int) -> _TrueOrNone: ...
+def snmp_set_counter64(__oid_num: int, __value: int) -> _TrueOrNone: ...
+def snmp_set_gauge(__oid_num: int, __value: int) -> _TrueOrNone: ...
+def snmp_incr_counter32(__oid_num: int, __increment: int) -> _TrueOrNone: ...
+def snmp_incr_counter64(__oid_num: int, __increment: int) -> _TrueOrNone: ...
+def snmp_incr_gauge(__oid_num: int, __increment: int) -> _TrueOrNone: ...
+def snmp_decr_counter32(__oid_num: int, __decrement: int) -> _TrueOrNone: ...
+def snmp_decr_counter64(__oid_num: int, __decrement: int) -> _TrueOrNone: ...
+def snmp_decr_gauge(__oid_num: int, __decrement: int) -> _TrueOrNone: ...
+@overload
+def send_to_spooler(__mesage_dict: dict[bytes, bytes]) -> bytes | None: ...
+@overload
+def send_to_spooler(
+    *, spooler: bytes = ..., priority: bytes = ..., at: bytes = ..., body: bytes = ..., **kwargs: bytes
+) -> bytes | None: ...
+
+spool = send_to_spooler
+
+def set_spooler_frequency(__frequency: int) -> Literal[True]: ...
+def spooler_jobs() -> list[bytes]: ...
+def spooler_pid() -> int: ...
+def spooler_pids() -> list[int]: ...
+def spooler_get_task(__task_path: str) -> dict[bytes, bytes] | None: ...
+def call(__rpc_name: str, **args: bytes) -> bytes | None: ...
+def chunked_read(__timeout: int = 0) -> bytes: ...
+def chunked_read_nb() -> bytes: ...
+def cl() -> int: ...
+def close(__fd: int) -> None: ...
+def connect(__socket_name: str, timeout: int = 0) -> int: ...
+def connection_fd() -> int: ...
+def disconnect() -> None: ...
+def embedded_data(__name: str) -> bytes: ...
+def extract(__name: str) -> bytes | None: ...
+def farm_get_msg() -> bytes | None: ...
+def farm_msg(__farm_name: str, __message: str | bytes | ReadOnlyBuffer) -> None: ...
+def get_logvar(__key: str | bytes) -> bytes | None: ...
+def green_schedule() -> Literal[True]: ...
+def i_am_the_lord() -> bool: ...
```

Comment:
I'm not going to check each function in detail, but I checked this one because the name sounded funny :D

I found [this doc](https://uwsgi-docs.readthedocs.io/en/latest/Legion.html) which says it should take a name, maybe like this:

```suggestion
def i_am_the_lord(__name: str) -> bool: ...
```

I think stubtest and stubgen don't work very well with some functions implemented in C, if the function doesn't have the right info about its parameters. Maybe check the docs for all functions that take no arguments?

Reply:
> I'm not going to check each function in detail, but I checked this one because the name sounded funny :D

I enjoyed the error message we got at the end of a stubtest run on https://github.com/python/typeshed/pull/10432/commits/39acdfe88af6948dfb96b2e9e1f66734a08c39bd:

```
note: unused allowlist entry uwsgi.i_am_the_lord
note: unused allowlist entry uwsgi.lord_scroll
note: unused allowlist entry uwsgi.scrolls
Found 3 errors (checked 2 modules)
worker 1 buried after 0 seconds
goodbye to uWSGI.
Sat Jul 22 18:26:07 2023 - uWSGI spooler 1947 screams: UAAAAAAH my master disconnected: i will kill myself !!!
```

Reply:
You are correct. That probably was my bad and I just skipped over the call to `PyArgParseTuple` when I was adding annotations (possibly because I was elated to see `Py_False`, rather than their weird `True` or `None` situation they've got going on everywhere). But I will double check that all the functions that take no arguments actually take no arguments, in case stubgen has some weird corner case there.

IIRC stubgen just punts and annotates everything with `*args, **kwargs` unless the docstring contains the callspec and then it will just trust that the docstring is correct, without checking it. It doesn't seem like it performs any static analysis on the C code, otherwise it would be possible to at least derive partial information from the method flags.